src/coord/src/persistence.rs

```diff
@@ -0,0 +1,250 @@
+                    id, e
+                );
+                self.disable();
+                return None;
+            }
+        };
+
+        if let Err(e) = self.wals.resume(id) {
+            error!(
+                "encountered error trying to resume write-ahead log for persisted table {} {}",
+                id, e
+            );
+            self.disable();
+            return None;
+        };
+
+        if let Err(e) = self
+            .compacter_tx
+            .send(CompacterMessage::Resume(id, persisted_trace))
+        {
+            debug!("compacter dropped, disabling WAL: {}", e);
+            self.disable();
+            return None;
+        }
+
+        messages
+    }
+
+    /// Write progress updates to all table WALs.
+    pub fn write_progress(&mut self, timestamp: Timestamp) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.write_progress(timestamp) {
+            error!(
+                "encountered error trying to write progress at ts {}: {}",
+                timestamp, e
+            );
+            self.disable();
+        }
+    }
+
+    /// Start persisting table `id`.
+    ///
+    /// Create a new WAL map entry and directory for `id` and tell the compacter
+    /// to start managing it.
+    pub fn create(&mut self, id: GlobalId) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.create(id) {
+            error!(
+                "encountered error creating table for relation {}: {}",
+                id, e
+            );
+            self.disable();
+            return;
+        }
+        if let Err(e) = self.compacter_tx.send(CompacterMessage::Add(id)) {
+            debug!("compacter dropped, disabling WAL: {}", e);
+            self.disable();
+        }
+    }
+
+    /// Stop persisting table `id`.
+    ///
+    /// Tell the Compacter to drop relation `id` and also delete the persisted WAL
+    /// and batch files. Also, remove the relation from the table from the in-memory
+    /// WAL map.
+    pub fn destroy(&mut self, id: GlobalId) {
+        if self.disabled {
+            return;
+        }
+
+        if let Err(e) = self.wals.destroy(id) {
+            error!(
+                "encountered error destroying table for relation {}: {}",
+                id, e
+            );
+            self.disable();
+            return;
+        }
+        if let Err(e) = self.compacter_tx.send(CompacterMessage::Drop(id)) {
+            debug!("compacter dropped, disabling WAL: {}", e);
+            self.disable();
+        }
+    }
+
+    /// Write updates to the WAL.
+    pub fn write(&mut self, id: GlobalId, updates: &[Update]) {
+        if self.disabled {
+            return;
+        }
+        if let Err(e) = self.wals.write(id, updates) {
+            error!(
+                "encountered error writing to table for relation {}: {}",
```

Comment:
Unfortunately the actual I/O error that prevented the write is lost in the process. I simulated an out-of-disk-space condition and the log reported the following:

```
2021-03-03T07:40:01.286372Z ERROR coord::persistence: encountered error writing to table for relation u1: failed to write to relation: u1 wal segment 0    
2021-03-03T07:40:01.286390Z ERROR coord::persistence: No additional records will be persisted    
```

"no space left on device" or something along those lines should have been present in the error message.

Reply:
The underlying error is now properly printed.